yarn-project/noir-contracts/src/contracts/noir-aztec3/src/private_call_stack_item.nr

```diff
@@ -59,28 +59,29 @@ impl PrivateCallStackItem {
                 // TODO handle the offsets as a variable incremented during extraction?
                 args_hash: fields[10],
                 return_values: arr_copy_slice(fields, [0; crate::abi::MAX_RETURN_VALUES], 11),
-                new_commitments: arr_copy_slice(fields, [0; crate::abi::MAX_NEW_COMMITMENTS], 15),
-                new_nullifiers: arr_copy_slice(fields, [0; crate::abi::MAX_NEW_NULLIFIERS], 19),
-                private_call_stack: arr_copy_slice(fields, [0; crate::abi::MAX_PRIVATE_CALL_STACK], 23),
-                public_call_stack: arr_copy_slice(fields, [0; crate::abi::MAX_PUBLIC_CALL_STACK], 27),
-                new_l2_to_l1_msgs:arr_copy_slice(fields, [0; crate::abi::MAX_L2_TO_L1_MSGS], 31),
+                read_requests: arr_copy_slice(fields, [0; crate::abi::MAX_READ_REQUESTS], 15),
+                new_commitments: arr_copy_slice(fields, [0; crate::abi::MAX_NEW_COMMITMENTS], 19),
+                new_nullifiers: arr_copy_slice(fields, [0; crate::abi::MAX_NEW_NULLIFIERS], 23),
+                private_call_stack: arr_copy_slice(fields, [0; crate::abi::MAX_PRIVATE_CALL_STACK], 27),
+                public_call_stack: arr_copy_slice(fields, [0; crate::abi::MAX_PUBLIC_CALL_STACK], 31),
+                new_l2_to_l1_msgs:arr_copy_slice(fields, [0; crate::abi::MAX_L2_TO_L1_MSGS], 35),
                 // TODO #588, relevant issue: https://github.com/AztecProtocol/aztec-packages/issues/588
                 // encrypted_logs_hash:arr_copy_slice(fields, [0; 2], 44),
                 // unencrypted_logs_hash:arr_copy_slice(fields, [0; 2], 46),
                 // encrypted_log_preimages_length: fields[47],
                 // unencrypted_log_preimages_length: fields[48],
-                historic_private_data_tree_root: fields[33],
-                historic_private_nullifier_tree_root: fields[34],
-                historic_contract_tree_root: fields[35],
-                historic_l1_to_l2_messages_tree_root: fields[36],
+                historic_private_data_tree_root: fields[37],
+                historic_private_nullifier_tree_root: fields[38],
+                historic_contract_tree_root: fields[39],
+                historic_l1_to_l2_messages_tree_root: fields[40],
                 contract_deployment_data: ContractDeploymentData {
-                    constructor_vk_hash : fields[37],
-                    function_tree_root : fields[38],
-                    contract_address_salt : fields[39],
-                    portal_contract_address : fields[40],
+                    constructor_vk_hash : fields[41],
+                    function_tree_root : fields[42],
+                    contract_address_salt : fields[43],
+                    portal_contract_address : fields[44],
```

Comment:
This index is 44 which would indicate 45 fields, but if we subtract 4 (for the contract address and function data at the start which are _not_ a part of the public inputs), we get size 41 which matches `PUBLIC_INPUTS_LENGTH` in `abi.nr`.